gnd/src/main/java/com/google/android/gnd/ui/projectselector/ProjectSelectorViewModel.java

```diff
@@ -16,78 +16,67 @@
 package com.google.android.gnd.ui.projectselector;
 
 import android.util.Log;
-
 import androidx.lifecycle.LiveData;
 import androidx.lifecycle.MutableLiveData;
-
 import com.google.android.gnd.repository.DataRepository;
 import com.google.android.gnd.repository.Resource;
-import com.google.android.gnd.rx.Result;
 import com.google.android.gnd.system.AuthenticationManager;
 import com.google.android.gnd.ui.common.AbstractViewModel;
 import com.google.android.gnd.vo.Project;
-
-import io.reactivex.Observable;
+import io.reactivex.Single;
 import io.reactivex.subjects.PublishSubject;
-
 import java.util.Collections;
 import java.util.List;
-
 import javax.inject.Inject;
 
 public class ProjectSelectorViewModel extends AbstractViewModel {
   private static final String TAG = ProjectSelectorViewModel.class.getSimpleName();
 
-  private final DataRepository dataRepository;
   private final MutableLiveData<Resource<List<Project>>> projectSummaries;
   private final PublishSubject<Integer> projectSelections;
   private final MutableLiveData<Project> activeProject;
   private final MutableLiveData<Throwable> activateProjectErrors;
-  private final Observable<Result<Project>> activeProjectStream;
 
   @Inject
   ProjectSelectorViewModel(DataRepository dataRepository, AuthenticationManager authManager) {
-    this.dataRepository = dataRepository;
     this.projectSummaries = new MutableLiveData<>();
     this.activeProject = new MutableLiveData<>();
     this.activateProjectErrors = new MutableLiveData<>();
     this.projectSelections = PublishSubject.create();
 
+    disposeOnClear(
+        projectSelections
+            .switchMapSingle(
+                idx ->
+                    dataRepository
+                        .activateProject(getProjectSummary(idx).getId())
+                        .doOnError(this::onActiveProjectError)
+                        .onErrorResumeNext(Single.never()))
```

Comment:
Not for this change, but is this potentially a place where we'll want to retry on error in the future?

If I'm envisioning this correctly, at present if this errors we'll log (and eventually show) an error then we'll wait idly until the user issues another project selection click to try again. Retrying once or twice, and showing some message to that effect, would be a small UX enhancement (at least when the user has a connection).

Reply:
Good idea in practice, but probably a "nice to have" vs lots of other "must haves" at the moment. Let's keep it in the back of our minds for future improvements!